### PR TITLE
move amp to https

### PR DESCRIPTION
**Pull request by johnduffell**
This moves amp to https - the canonical and amphtml links need to be correct and we shouldn't rely on redirects like we do in /info

Also look at https://github.com/guardian/fastly-edge-cache/pull/488

In the process I found LinkTo was like this
image
so I made it more like this
image

I also found a load of code to rewrite all links on the page to have ?switchOn=dsfsdf&switchOff=sdflkhsd f you're running in dev mode to play with switches locally.  I'm sure it's just cluttering the code so I dealt with it.

@gklopper @NataliaLKB

### CODE REVIEW

File: common/app/common/LinkTo.scala [hunk 1]
```diff
@@ -29,42 +29,42 @@ trait LinkTo extends Logging {
   def apply(html: Html)(implicit request: RequestHeader): String = this(html.toString(), Edition(request))
   def apply(link: String)(implicit request: RequestHeader): String = this(link, Edition(request))
 
-  def apply(url: String, edition: Edition)(implicit request: RequestHeader): String = {
-    val processedUrl: String = processUrl(url, edition).url
-    handleQueryStrings(processedUrl)
-  }
+  def apply(url: String, edition: Edition)(implicit request: RequestHeader): String =
```
Comment (kelvin-chappell) [inline common/app/common/LinkTo.scala]: Some of this code is gu-specific and some of it is generic URL handling (eg path finding and protocol changing).  Would it be possible to abstract out the generic parts, possibly to implicits.Requests?

**Comment by gklopper** [comment]
:+1:  nice ratio...
ratio

File: common/app/common/LinkTo.scala (hunk 1 above)
Comment (johnduffell) [inline common/app/common/LinkTo.scala]: thanks for the comment and it's definitely not perfect, next time I'm tweaking this code I'll try to do that

**Comment by gklopper** [comment]
I'm looking forward to this shipping

**Comment by NataliaLKB** [comment]
Yay!

**Pull request merged**